test/jasmine/tests/geo_test.js

```diff
@@ -1811,85 +1980,145 @@ describe('Test geo zoom/pan/drag interactions:', function() {
+
+            plot(fig).then(function() {
+                _assert('base', [
+                    [undefined, 0], [undefined, undefined], undefined
+                ], [
+                    [-180, -0], [350, 260], [0, 0], 114.59
+                ], undefined);
+                return drag({path: [[350, 250], [400, 250]], noCover: true});
+            })
+            .then(function() {
+                _assert('after east-west drag', [
+                    [149.40, 0], [149.40, 0], 1.1249
+                ], [
+                    [-149.40, 0], [350, 260], [0, 0], 114.59
+                ], [
+                    'geo.projection.rotation.lon', 'geo.center.lon', 'geo.center.lat',
+                    'geo.projection.scale', 'geo.fitbounds'
+                ]);
+                return scroll([200, 250], [-200, -200]);
+            })
+            .then(function() {
+                _assert('after off-center scroll', [
+                    [127.176, 0], [127.176, 1.21], 1.484
+                ], [
+                    [-127.176, 0], [350, 263.195], [0, 0], 151.20
+                ], [
+                    'geo.projection.rotation.lon', 'geo.center.lon',
+                    'geo.center.lat', 'geo.projection.scale'
+                ]);
+                return Plotly.relayout(gd, 'geo.showocean', false);
+            })
+            .then(function() {
+                _assert('after some relayout call that causes a replot', [
+                    // converts translate (px) to center (lonlat)
+                    [127.176, 0], [127.176, 1.21], 1.484
+                ], [
+                    [-127.176, 0], [350, 260], [0, 1.21], 151.20
+                ], [
+                    'geo.showocean'
+                ]);
+                return dblClick([350, 250]);
+            })
+            .then(function() {
+                _assert('after double click', [
+                    [undefined, 0], [undefined, undefined], undefined
+                ], [
+                    [-180, -0], [350, 260], [0, 0], 114.59
+                ], 'dblclick');
+            })
+            .catch(failTest)
+            .then(done);
+        });
     });
 
-    it('should work for clipped projections', function(done) {
-        var fig = Lib.extendDeep({}, require('@mocks/geo_orthographic'));
-        fig.layout.dragmode = 'pan';
+    describe('should work for clipped projections', function() {
+        var fig;
 
-        // of layout width = height = 500
+        beforeEach(function() {
+            fig = Lib.extendDeep({}, require('@mocks/geo_orthographic'));
+            fig.layout.dragmode = 'pan';
+
+            // of layout width = height = 500
+        });
 
         function _assert(step, attr, proj, eventKeys) {
             var msg = '[' + step + '] ';
 
             var geoLayout = gd._fullLayout.geo;
-            var rotation = geoLayout.projection.rotation;
+            var rotation = geoLayout.projection.rotation || [];
```

Comment:
The default here should be an array or an object?

Reply:
Good eye. It's an object.

Reply:
done in https://github.com/plotly/plotly.js/pull/4419/commits/0e50599667c8350226c569b416ce631e01d5ed03